contrib/inventory/apstra_aos.py

```diff
@@ -0,0 +1,395 @@
+    }
+  },
+  "all": {
+    "hosts": [
+      "5254001CAFD8",
+      "52540073956E",
+      "525400DDDF72",
+      "525400E5486D",
+      "52540022211A"
+    ],
+    "vars": {}
+  },
+  "vEOS": {
+    "hosts": [
+      "5254001CAFD8",
+      "525400DDDF72"
+    ],
+    "vars": {}
+  }
+}
+"""
+class AosInventory(object):
+
+    def __init__(self):
+
+        """ Main execution path """
+
+        if not HAS_AOS_PYEZ:
+            raise Exception('aos-pyez is not installed.  Please see details here: https://github.com/Apstra/aos-pyez')
+
+        # Initialize inventory
+        self.inventory = dict()  # A list of groups and the hosts in that group
+        self.inventory['_meta'] = dict()
+        self.inventory['_meta']['hostvars'] = dict()
+
+        # Read settings and parse CLI arguments
+        self.read_settings()
+        self.parse_cli_args()
+
+        #----------------------------------------------------
+        # Open session to AOS
+        #----------------------------------------------------
+
+        aos = Session(self.aos_server)
+        # aos = Session(  server=self.aos_server,
+        #                 port=self.aos_server_port,
+        #                 user=self.aos_username,
+        #                 passwd=self.aos_password)
+
+        aos.login()
+
+        #----------------------------------------------------
+        # Build the inventory
+        #----------------------------------------------------
+        for device in aos.Devices:
+            # If not reacheable, create by key and
+            # If reacheable, create by hostname
+
+            self.add_host_to_group('all', device.name)
+
+            # populate information for this host
+            if 'status' in device.value.keys():
+                for key, value in device.value['status'].items():
+                    self.add_var_to_host(device.name, key, value)
+
+            if 'user_config' in device.value.keys():
+                for key, value in device.value['user_config'].items():
+                    self.add_var_to_host(device.name, key, value)
+
+            ## Based on device status online|offline, collect facts as well
+            if device.value['status']['comm_state'] == 'on':
+
+                # Populate variables for this host
+                self.add_var_to_host(device.name,
+                                     'ansible_ssh_host',
+                                     device.value['facts']['mgmt_ipaddr'])
+
+                # self.add_host_to_group('all', device.name)
+                for key, value in device.value['facts'].items():
+                    self.add_var_to_host(device.name, key, value)
+
+                    if key == 'os_family':
+                        self.add_host_to_group(value, device.name)
+                    elif key == 'hw_model':
+                        self.add_host_to_group(value, device.name)
+
+            ## Check if device is associated with a blueprint
+            ##   if it's create a new group
```

Comment:
pep8/flake8 doesnt really like the '##' comments. Minor, but it keeps pep8 happy:
contrib/inventory/apstra_aos.py:286:9: E265 block comment should start with '# '
contrib/inventory/apstra_aos.py:288:9: E265 block comment should start with '# '
contrib/inventory/apstra_aos.py:298:9: E265 block comment should start with '# '
contrib/inventory/apstra_aos.py:300:9: E265 block comment should start with '# '
contrib/inventory/apstra_aos.py:316:13: E266 too many leading '#' for block comment
contrib/inventory/apstra_aos.py:333:13: E266 too many leading '#' for block comment
contrib/inventory/apstra_aos.py:334:13: E266 too many leading '#' for block comment
contrib/inventory/apstra_aos.py:342:9: E265 block comment should start with '# '
contrib/inventory/apstra_aos.py:344:9: E265 block comment should start with '# '